lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -866,7 +866,8 @@ bool TypeLoweringVisitor::visitStmt(RefConnectOp op) {
   for (const auto &field : llvm::enumerate(fields)) {
     Value src = getSubWhatever(op.getSrc(), field.index());
     Value dest = getSubWhatever(op.getDest(), field.index());
-    builder->create<RefConnectOp>(dest, src);
+    assert(!field.value().isOutput);
```

Comment:
Like above, will fail with flips and aggregates.

Reply:
I'll check, but the idea was to just be defensive about the lack of the `std::swap` in other connects, since references should never have flips within them.  So if we (not planned) add flips this code will assert before doing the wrong thing.

Maybe should've reverted first to avoid this confusion, but that seemed unnecessarily aggressive since the two are largely the same / implemented similarly, so went with swapping in an improved version :smile: , but seems I missed some aux changes along the way.  Anyway, this portion of the diff and generally is more easily read if you skip the revert portion, this method is entirely new if that helps.